api/files/serializers.py

```diff
@@ -423,6 +429,57 @@ def get_download_link(self, obj):
             view_only=self.context['request'].query_params.get('view_only'),
         )
 
+class FileMetadataRecordSerializer(JSONAPISerializer):
+
+    id = IDField(read_only=True, source='_id')
+    type = TypeField()
+
+    metadata = ser.DictField()
+
+    file = RelationshipField(
+        related_view='files:file-detail',
+        related_view_kwargs={'file_id': '<file._id>'},
+    )
+
+    schema = RelationshipField(
+        related_view='schemas:file-metadata-schema-detail',
+        related_view_kwargs={'schema_id': '<schema._id>'},
+    )
```

Comment:
Very good point. Hmm! I think this requires a bit more discussion/thinking, adding it to my list of things to ponder.

Reply:
OK! Had a good long think about this one. How about this as a proposed solution?

So the `schema` relationship is really referring to the `schema` that's stored on the FileMetadataRecord, and the `metadata` is to the `metadata` field on the model, which is what's being updated by a user (the schema is never updated by the user at all, only metadata).

I realized the problem really is that `validate_metadata` on the serializer does its thing and only returns vague errors, and that what it's validating against isn't really transparent. The user entered metadata schema is stored on the record's serializer (different from API serializers, [they can be found here](https://github.com/CenterForOpenScience/osf.io/pull/8721/files#diff-898ca0bd378d7ba622a0e1b2c78f2abfR34)  in the `osf_schema` property. So what if updating the `metadata` field just returned the schema it's validating against as part of the error? Something like:

```
{
    "source": {},
    "meta": {},
    "detail": {
        "message": "Your response of dinosaur for the field related_publication_doi was invalid.",
        "metadata_schema": {
            "description": "This is validation for file metadata fields that the OSF allows to be edited by the user. These fields will match the fields presented to the user on a file metadata form on the OSF.",
            "title": "OSF Specific Fields for the Datacite Schema",
            "$id": "/osf_datacite",
            "additionalProperties": "False",
            "$schema": "http://json-schema.org/draft-07/schema#",
            "type": "object",
            "properties": {
                "related_publication_doi": {
                    "pattern": "10\\..+/.+",
                    "type": "string",
                    "description": "The DOI of a related resource."
                },
                "funders": {
                    "items": {
                        "type": "object",
                        "properties": {
                            "funding_agency": {
                                "type": "string",
                                "description": "The name of the funding provider"
                            },
                            "grant_number": {
                                "type": "string",
                                "description": "The code assigned by the funder to a sponsored award (grant)."
                            }
                        }
                    },
                    "type": "array",
                    "description": "Funders responsible for funding the resource."
                },
                "file_description": {
                    "type": "string",
                    "description": "The description of the file"
                },
                "resource_type": {
                    "enum": [
                        "Audio/Video",
                        "Dataset",
                        "Image",
                        "Model",
                        "Software",
                        "Book",
                        "Funding Submission",
                        "Journal Article",
                        "Lesson",
                        "Poster",
                        "Preprint",
                        "Presentation",
                        "Research Tool",
                        "Thesis",
                        "Other"
                    ],
                    "type": "string"
                }
            }
        }
    }
}
```

That way, we're not re-naming fields on a serializer to be different than how they're stored on the model, and the user is still getting feedback on what fields are expected to go into the `metadata` field. 

@pattisdr @sloria what do you think?

Reply:
I think this is a smart idea @erinspace - As you pointed out, osf_datacite.json is the schema for the `metadata` field only, not the schema for the FileMetadataRecord as a whole, which we are serializing here.   I think it makes a lot of sense to return it as part of the error message, so the user can figure out what is wrong with their request, if needed, while still properly serializing the FileMetdataRecord. Win-win!

Reply:
I'm also good with that. Maybe add `metadata_schema` to `meta` in the error object so that it complies with JSON API: https://jsonapi.org/format/#error-objects